cleverhans/future/torch/attacks/deepfool.py

```diff
@@ -0,0 +1,86 @@
+"""
+DeepFool Attack from https://arxiv.org/pdf/1511.04599.pdf 
+by Seyed Mohsen et al.
+"""
+
+import torch
+import torch.nn as nn
+import numpy as np
+import copy
+
+from torch.autograd.gradcheck import zero_gradients as zgrad
+
+
+__all__ = ['deepfool']
+
+
+def deepfool(x, net:nn.Module, eta:float=0.02, tnc:int=10, miter:int=50, 
+             mret=False):
+  """
+  Implementation of the DeepFool Attack.
+
+  Arguments
+  ---------
+  x : torch.Tensor
+      The input tensor
+  net : nn.Module
+        The net.
+  eta : float
+        Eta as used in the paper.
+  tnc : int
+       The number of classes to consider.
+  miter : int
+          The maximum number of iterations.
+  mret : bool
```

Comment:
DeepFool also says that the total perturbation it causes to an image can be used to measure the robustness of the model, so I have allowed this extra parameter `mret` which if true (defaults to `False`) returns that metric along with the adversarial image.

Reply:
Empirically measuring robustness in this way is not specific to deepfool so I think to remain consistent with other attacks, we'll only return the adversarial example and if users want to measure robustness in this way they can diff the original and adversarial inputs